fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/LuceneOptimizedCompoundReader.java

```diff
@@ -0,0 +1,197 @@
+            try {
+                CodecUtil.checkIndexHeader(entriesStream, LuceneOptimizedCompoundFormat.ENTRY_CODEC,
+                         LuceneOptimizedCompoundFormat.VERSION_START,
+                         LuceneOptimizedCompoundFormat.VERSION_CURRENT, segmentID, "");
+                final int numEntries = entriesStream.readVInt();
+                mapping = new HashMap<>(numEntries);
+                for (int i = 0; i < numEntries; i++) {
+                    final FileEntry fileEntry = new FileEntry();
+                    final String id = entriesStream.readString();
+                    FileEntry previous = mapping.put(id, fileEntry);
+                    if (previous != null) {
+                        throw new CorruptIndexException("Duplicate cfs entry id=" + id + " in CFS ", entriesStream);
+                    }
+                    fileEntry.offset = entriesStream.readLong();
+                    fileEntry.length = entriesStream.readLong();
+                }
+            } catch (Throwable exception) {
+                priorE = exception;
+            } finally {
+                CodecUtil.checkFooter(entriesStream, priorE);
+            }
+        }
+        return Collections.unmodifiableMap(mapping);
+    }
+
+    @Override
+    public void close() throws IOException {
+        IOUtils.close(handle);
+    }
+
+    @Override
+    public IndexInput openInput(String name, IOContext context) throws IOException {
+        ensureOpen();
+        final String id = IndexFileNames.stripSegmentName(name);
+        final FileEntry entry = entries.get(id);
+        if (entry == null) {
+            String datFileName = IndexFileNames.segmentFileName(segmentName, "", LuceneOptimizedCompoundFormat.DATA_EXTENSION);
+            throw new FileNotFoundException("No sub-file with id " + id + " found in compound file \"" + datFileName + "\" (fileName=" + name + " files: " + entries.keySet() + ")");
+        }
+        return handle.slice(name, entry.offset, entry.length);
+    }
+
+    /** Returns an array of strings, one for each file in the directory. */
+    @Override
+    public String[] listAll() {
+        ensureOpen();
+        String[] res = entries.keySet().toArray(new String[entries.size()]);
+
+        // Add the segment name
+        for (int i = 0; i < res.length; i++) {
+            res[i] = segmentName + res[i];
+        }
+        return res;
+    }
+
+    /** Not implemented.
+     * @throws UnsupportedOperationException always: not supported by CFS */
+    @Override
+    public void deleteFile(String name) {
+        throw new UnsupportedOperationException();
+    }
+
+    /** Not implemented.
+     * @throws UnsupportedOperationException always: not supported by CFS */
+    @Override
+    public void rename(String from, String to) {
+        throw new UnsupportedOperationException();
+    }
+
+    /** Not implemented. */
+    @Override
+    public void syncMetaData() {
+        // Not implemented
```

Comment:
It might be good in this comment to explain why it's not implemented (and ideally why it's fine that this doesn't throw, unlike the other unimplemented functions)

Reply:
Actually I am not sure why this is empty rather than throwing unsupported exception. The original implementation ```Lucene50CompoundReader``` just does same thing here. We want to follow what it does. I guess this is called by somewhere that we don't want to break with throwing exceptions. But the ```syncMetaData```, which is to ```Ensures that directory metadata, such as recent file renames, are moved to stable storage.```, is not meaningful for compound reader. Probably @jleach4 has better idea for this?